### PR TITLE
added Hug command

### PR DESCRIPTION
**Pull request by namanyt**
`+hug` will send a embed with a hug embed as shown in the image

Screen Shot 2021-05-07 at 10 45 27 AM

`+huglb` will send this leaderboard type embed 

Screen Shot 2021-05-07 at 10 45 16 AM

this for #64

(you cannot hug yourself, `congratulations, you played yourself`)

### CODE REVIEW

**Comment by namanyt** [comment]
this is for a issue#64

File: .gitignore [hunk 1]
```diff
@@ -1,6 +1,7 @@
 ### Custom ###
 dynamic/
 .env
+.vscode/
```
Comment (mshaugh) [inline .gitignore:4]: Do not change the `.gitignore`

File: carberretta/bot/cogs/misc.py [hunk 2]
```diff
@@ -4,18 +4,92 @@
 A place for commands which don't fit anywhere else.
 """
 
+import datetime
 import unicodedata
 
 import discord
 from discord.ext import commands
+from discord.ext.commands import BucketType
```
Comment (mshaugh) [inline carberretta/bot/cogs/misc.py:12]: `discord.ext.commands` is already imported, no need to import BucketType here.

File: carberretta/bot/cogs/misc.py [hunk 3]
```diff
@@ -4,18 +4,92 @@
...
+from discord.ext.menus import MenuPages, ListPageSource
 
 from carberretta.utils import DEFAULT_EMBED_COLOUR
 
 
+class Leaderboard(ListPageSource):
```
Comment (mshaugh) [inline carberretta/bot/cogs/misc.py:18]: Bot does not need to be passed in here, it is never used.
Remove `write_page`, and just use `format_page` to do all of the work.
Prefer `Embed.from_dict`.
Use type hints.

File: carberretta/bot/cogs/misc.py [hunk 4]
```diff
@@ -4,18 +4,92 @@
...
+        return await self.write_page(menu, offset, fields)
+
+
 class Miscellaneous(commands.Cog):
     def __init__(self, bot: commands.Bot) -> None:
         self.bot = bot
 
+    @commands.command(name="hug", aliases=["huggies", "hugs", "huggie"])
+    @commands.cooldown(1, 10, BucketType.user)
+    async def hug_user_command(self, ctx: commands.Context, user: discord.Member):
```
Comment (mshaugh) [inline carberretta/bot/cogs/misc.py:53]: Adhere to PEP8 standards.
Variable names need to be more meaningful.
Prefer `command_x` naming scheme.
Prefer `Embed.from_dict`.
Use type hints.

File: carberretta/data/static/build.sql [hunk 5]
```diff
@@ -17,3 +17,9 @@ CREATE TABLE IF NOT EXISTS premieres (
     Upcoming int,
     Announced int
 );
+
+CREATE TABLE IF NOT EXISTS hugLeaderboard (
```
Comment (mshaugh) [inline carberretta/data/static/build.sql:21]: Call the table `hugs`
Prefer `given` and `received` to `TimesHuggedOthers` and `TimesGotHugs`.

**Review by mshaugh — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 1 above)
Comment (namanyt) [reply .gitignore:4]: I actually did it cus my vscode was sending notification for adding this, as I added to gitignore to ignore it

Comment (parafoxia) [reply .gitignore:4]: Just don't commit the folder.

Comment (namanyt) [reply .gitignore:4]: ok

Comment (mshaugh) [reply .gitignore]: Use a "global" gitignore. See https://git-scm.com/docs/gitignore and `core.excludesFile`.

File: carberretta/bot/cogs/misc.py [hunk 6]
```diff
@@ -4,18 +4,88 @@
...
+            return
+
+        if times_hugged_others_2 is None:
+            await self.bot.db.execute('INSERT INTO hugs (UserID) VALUES (?)', user_id)
+        if times_hugged_others is None:
+            await self.bot.db.execute('INSERT INTO hugs (UserID) VALUES (?)', user_id_2)
+
+        await self.bot.db.execute(
+            'UPDATE hugLeaderboard SET given = given + 1 WHERE UserID = ?', user_id_2)
+        await self.bot.db.execute('UPDATE hugLeaderboard SET received = received + 1 WHERE UserID = ?', user_id)
```
Comment (parafoxia) [inline carberretta/bot/cogs/misc.py:68]: Combine SQL statements.

File: carberretta/bot/cogs/misc.py [hunk 7]
```diff
@@ -4,18 +4,88 @@
...
+        received_count = await self.bot.db.record('SELECT received FROM hugs WHERE UserID = ?', user_id)
+        await self.bot.db.commit()
```
Comment (parafoxia) [inline carberretta/bot/cogs/misc.py:70]: Not needed.

File: carberretta/data/static/build.sql [hunk 8]
```diff
@@ -17,3 +17,9 @@ CREATE TABLE IF NOT EXISTS premieres (
     Upcoming int,
     Announced int
 );
+
+CREATE TABLE IF NOT EXISTS hugs (
+    UserID INTEGER NOT NULL PRIMARY KEY,
+    received INTEGER DEFAULT 0,
+    given INTEGER DEFAULT 0
```
Comment (parafoxia) [inline carberretta/data/static/build.sql:24]:
```suggestion
    Received INTEGER DEFAULT 0,
    Given INTEGER DEFAULT 0
```

**Review by parafoxia — CHANGES_REQUESTED** [review]

File: carberretta/bot/cogs/misc.py (hunk 7 above)
Comment (namanyt) [reply carberretta/bot/cogs/misc.py:70]: It is needed as I found it was not committing anything to the data base from this cog

Comment (parafoxia) [reply carberretta/bot/cogs/misc.py:70]: It should automatically commit using the scheduled task. It is does not, you need to look into why.

File: carberretta/data/static/build.sql (hunk 8 above)
Comment (namanyt) [reply carberretta/data/static/build.sql]: ok

**Review by parafoxia — CHANGES_REQUESTED** [review]
The single change you've made is fine, but there are still multiple outstanding comments that have not been resolved.

**Pull request closed without merging**